ui/app/routes/exec.js

```diff
@@ -17,12 +17,20 @@ export default Route.extend(WithWatchers, {
     const namespace = transition.to.queryParams.namespace || this.get('system.activeNamespace.id');
     const name = params.job_name;
     const fullId = JSON.stringify([name, namespace || 'default']);
-    return this.store
+
+    const jobPromise = this.store
       .findRecord('job', fullId)
       .then(job => {
         return job.get('allocations').then(() => job);
       })
       .catch(notifyError(this));
+
+    return Promise.all([jobPromise, import('xterm').then(module => module.Terminal)]);
```

Comment:
The async import is definitely sneaky tucked into a model hook like this. I looked right past it at first.

I think lifecycle-wise this is the right place to put it, especially in parallel with the the job promise.

I simultaneously want to hide the import details in some util or service and want to shy away from abstracting language-level features like this.

Maybe pulling the import onto its own line will make it at least standout more?

```js
const importXterm = import('xterm').then(({ Terminal }) => Terminal)
return Promise.all([jobPromise, importXterm]);
```

Dunno 🤷‍♂️It's definitely uncharted territory for me.

Reply:
I did make the job promise its own variable and I agree that it makes sense to make the import one too, now done, thanks. We might not do much of this but if we do have more maybe we can come up with something more general!